static/src/server.ts

```diff
@@ -249,68 +339,140 @@ function getBarTileSvg(
+      };
+    })
+    .catch(() => {
+      console.log("Failed to get bar tile result for: " + id);
+      return null;
+    });
 }
 
-// Gets the svg element for a map tile
-function getMapTileSvg(
+// Gets the TileResult for a map tile
+function getMapTileResult(
+  id: string,
   tileConfig: TileConfig,
   place: NamedTypedPlace,
   enclosedPlaceType: string,
   statVarSpec: StatVarSpec
-): Promise<SVGSVGElement> {
+): Promise<TileResult> {
   const tileProp = {
-    id: DOM_ID,
+    id,
     title: tileConfig.title,
     place,
     enclosedPlaceType,
     statVarSpec,
     svgChartHeight: SVG_HEIGHT - LEGEND_MARGIN_VERTICAL * 2,
     apiRoot: CONFIG.apiRoot,
   };
-  return fetchMapData(tileProp).then((chartData) => {
-    const legendContainer = document.createElement("div");
-    const mapContainer = document.createElement("div");
-    drawMap(
-      chartData,
-      tileProp,
-      null,
-      legendContainer,
-      mapContainer,
-      SVG_WIDTH
-    );
-    // Get the width of the text in the legend
-    let legendTextWidth = 0;
-    Array.from(legendContainer.querySelectorAll("text")).forEach((node) => {
-      legendTextWidth = Math.max(node.getBBox().width, legendTextWidth);
+  return fetchMapData(tileProp)
+    .then((chartData) => {
+      const legendContainer = document.createElement("div");
+      const mapContainer = document.createElement("div");
+      drawMap(
+        chartData,
+        tileProp,
+        null,
+        legendContainer,
+        mapContainer,
+        SVG_WIDTH
+      );
+      // Get the width of the text in the legend
+      let legendTextWidth = 0;
+      Array.from(legendContainer.querySelectorAll("text")).forEach((node) => {
+        legendTextWidth = Math.max(node.getBBox().width, legendTextWidth);
+      });
+      const legendWidth = legendTextWidth + MAP_LEGEND_CONSTANT_WIDTH;
+      // Create a single merged svg to hold both the map and the legend svgs
+      const mergedSvg = document.createElementNS(
+        "http://www.w3.org/2000/svg",
+        "svg"
+      );
+      mergedSvg.setAttribute("height", String(SVG_HEIGHT));
+      mergedSvg.setAttribute("width", String(SVG_WIDTH));
+      // Get the map svg and add it to the merged svg
+      const mapSvg = mapContainer.querySelector("svg");
+      const mapWidth = SVG_WIDTH - legendWidth;
+      mapSvg.setAttribute("width", String(mapWidth));
+      const mapG = document.createElementNS("http://www.w3.org/2000/svg", "g");
+      mapG.appendChild(mapSvg);
+      mergedSvg.appendChild(mapG);
+      // Get the legend svg and add it to the merged svg
+      const legendSvg = legendContainer.querySelector("svg");
+      legendSvg.setAttribute("width", String(legendWidth));
+      const legendG = document.createElementNS(
+        "http://www.w3.org/2000/svg",
+        "g"
+      );
+      legendG.setAttribute("transform", `translate(${mapWidth})`);
+      legendG.appendChild(legendSvg);
+      mergedSvg.appendChild(legendG);
```

Comment:
I wonder can the merge happen in the original map component? I wonder if this merge has to happen in multiple places now?

Reply:
- doing it as a "merge" (draw legend and map and then merge into one svg) doesn't make the most sense because it is forcing a redraw that will sometimes be unnecessary & will need some thought to account for resizing.
- doing it as just drawing legend and map into the same container will need some thought to ensure the drawing functions don't break other map components (e.g., map explorer we would want to keep the legend and actual map separate because we want to zoom the map without affecting the legend)

So I think this could be a future thing & I could open an issue for it?

Reply:
I am not fully sure why option 1 is a redraw since it just operates on html element. But ok to solve it later. Worth to add where this merging happens (I guess here and in chart embed?)

Reply:
ah you're right, no need to redraw, just need to manipulate the html elements. But will take more thought especially around calculating BBox (which the chart embed merge function uses to determine where to move each element).

created issue: https://github.com/datacommonsorg/website/issues/2808